packages/app/src/server/service/passport.ts

```diff
@@ -719,8 +722,40 @@ class PassportService implements S2sMessageHandlable {
       return response.statusCode === 200;
     }
     catch (err) {
-      logger.debug('Issuer host unreachable:', err.code);
+      logger.error('OidcStrategy: issuer host unreachable:', err.code);
+    }
+  }
+
+  /**
+   * Get oidcIssuer object
+   * Utilize p-retry package to retry oidcIssuer initialization 3 times
+   *
+   * @param issuerHost
+   * @returns instance of OIDCIssuer
+   */
+  async getOIDCIssuerInstace(issuerHost) {
+    const oidcIssuerHostReady = await this.isOidcHostReachable(issuerHost);
+    if (!oidcIssuerHostReady) {
+      logger.error('OidcStrategy: setup failed: OIDC Issur host unreachable');
+      return;
     }
+    const oidcIssuer = await pRetry(async() => {
+      return OIDCIssuer.discover(issuerHost);
+    }, {
+      onFailedAttempt: (error) => {
+        if (error.attemptNumber === 1) {
+          logger.debug('OidcStrategy: setup failed. Retrying ...');
+        }
+        else {
+          logger.debug(`OidcStrategy: setup attempt ${error.attemptNumber - 1} failed. Retrying ...`);
+        }
+
+      },
+      retries: 3,
+    }).catch(() => {
```

Comment:
Please include the error object you catch in the log message. It will help us to solve problems in the future.